cookbook/tests/test_core.py

```diff
@@ -0,0 +1,185 @@
+
+    def fetch_execute_task(self):
+        remote = FlyteRemote()
+        flyte_task = remote.fetch_task(PROJECT, DOMAIN, self.name, VERSION)
+        execution = remote.execute(flyte_task, self.input)
+        execution.wait_for_completion()
+        assert execution.outputs["o0"] == self.output
+
+    def monitor_workflow_execution(self):
+        remote = FlyteRemote()
+        flyte_launch_plan = remote.fetch_launch_plan(
+            PROJECT, DOMAIN, self.name, VERSION
+        )
+        execution = remote.execute(flyte_launch_plan, self.input)
+
+        poll_interval = datetime.timedelta(seconds=1)
+        time_to_give_up = datetime.datetime.utcnow() + datetime.timedelta(seconds=60)
+
+        execution.sync()
+        while datetime.datetime.utcnow() < time_to_give_up:
+
+            if execution.is_complete:
+                execution.sync()
+                break
+
+            with pytest.raises(
+                FlyteAssertion, match="Please wait until the node execution has completed before requesting the outputs"
+            ):
+                execution.outputs
+
+            time.sleep(poll_interval.total_seconds())
+            execution.sync()
+
+            if execution.node_executions:
+                assert execution.node_executions["start-node"].closure.phase == 3  # SUCCEEEDED
+
+        for key in execution.node_executions:
+            assert execution.node_executions[key].closure.phase == 3
+
+        assert execution.node_executions["n0"].inputs == self.input
+        assert execution.node_executions["n0"].outputs["o0"] == self.output
+        assert execution.node_executions["n0"].task_executions[0].inputs == self.input
+        assert execution.node_executions["n0"].task_executions[0].outputs["o0"] == self.output
+        assert execution.inputs == self.input
+        assert execution.outputs["o0"] == self.output
+
+
+@pytest.fixture(scope="session")
+def flyte_workflows_register(request):
+    subprocess.check_call(
+        f"flytectl register example -p {PROJECT} -d {DOMAIN} --version={VERSION}",
+        shell=True,
+    )
+
+
+def get_test_manifest():
+    f = open(os.path.join(os.path.dirname(__file__), FLYTE_TEST_MANIFEST_PATH))
+    return Manifest(json.load(f))
+
+
+def join_thread(threads_list):
+    for t in threads_list:
+        t.join()
+
+
+def filter_results(test_test: str):
+    results = []
+
+    for example in get_test_manifest().manifest:
+        if example["name"] == "core":
+            for name in example["tests"]:
+                manifest = TestManifest(name, example['tests'][name])
+                if manifest.type == test_test:
+                    results.append(manifest)
+
+    return results
+
+
+def test_fetch_execute_task(flyteclient, flyte_workflows_register):
+    threads_list = list()
+
+    for manifest in filter_results("task"):
+        print(f"Running {manifest.type} for {manifest.name}")
+        t = threading.Thread(target=manifest.fetch_execute_task, args=())
```

Comment:
I think we should use a threadpool executor instead of this. This will lead to many parallel executions, which may not be possible on sandbox? 

Maybe we should use - https://docs.python.org/3/library/asyncio-task.html